app/src/main/java/com/example/footballmanager/screens/result_screens/success_components/MatchScoreTab.kt

```diff
@@ -0,0 +1,324 @@
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.em
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import coil.compose.rememberAsyncImagePainter
+import com.example.footballmanager.R
+import com.example.footballmanager.network.FixtureDataWrapper
+import com.example.footballmanager.screens.view_models.HomeViewModel
+import java.time.LocalDate
+import kotlin.math.log
+
+
+@Composable
+fun MatchScoreTab(
+    fixture: FixtureDataWrapper,
+    modifier: Modifier = Modifier,
+    homeViewModel: HomeViewModel = viewModel()
+) {
+    var lastLoadedAmount by remember {
+        mutableIntStateOf(10)
+    }
+    val lazyListState = rememberLazyListState()
+    val currentState = remember { derivedStateOf { lazyListState.firstVisibleItemIndex } }
+    val shouldLoadMore = remember { derivedStateOf { currentState.value + 10 >= lastLoadedAmount } }
+
+    LaunchedEffect(key1 = shouldLoadMore.value) {
+        lastLoadedAmount += 10
+    }
+
+    Column(
+        verticalArrangement = Arrangement.spacedBy(
+            dimensionResource(id = R.dimen.column_spacing),
+            Alignment.Top
+        ),
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(horizontal = dimensionResource(id = R.dimen.column_padding))
+    ) {
+        val defaultValue = stringResource(id = R.string.default_value)
+        LazyColumn(
+            state = lazyListState,
+            modifier = Modifier
+                .fillMaxWidth()
+        ) {
+            var previousLeagueName = defaultValue
+            for (i in 0..lastLoadedAmount) {
+                val item = fixture.responseBody[i]
+                val leagueName = item.league?.name ?: defaultValue
+                val leagueLogo = item.league?.logo ?: defaultValue
+                val nameTeamHome = item.teams?.home?.name ?: defaultValue
+                val nameTeamAway = item.teams?.away?.name ?: defaultValue
+                val scoreTeamHome = item.goals?.home?.toString() ?: defaultValue
+                val scoreTeamAway = item.goals?.away?.toString() ?: defaultValue
+                val logoTeamHome = item.teams?.home?.logo ?: defaultValue
+                val logoTeamAway = item.teams?.away?.logo ?: defaultValue
+                val short = item.fixture?.status?.short ?: defaultValue
+
+                if (leagueName != previousLeagueName) {
```

Comment:
I don't get it. Why are you comparing these values?

Reply:
It is like that to avoid having the same league name twice. The operation ensures that if matches that are present in the same league appear next to each other, the league name displays only once (not above every match)